base/namedtuple.jl

```diff
@@ -137,9 +137,13 @@ function show(io::IO, t::NamedTuple)
     if n == 0
         print(io, "NamedTuple()")
     else
+        typeinfo = get(io, :typeinfo, Any)
         print(io, "(")
         for i = 1:n
-            print(io, fieldname(typeof(t),i), " = "); show(io, getfield(t, i))
+            print(io, fieldname(typeof(t),i), " = ")
+            show(IOContext(io, :typeinfo =>
+                           t isa typeinfo <: NamedTuple ? fieldtype(typeinfo, i) : Any),
```

Comment:
This should use the field type directly from `t` if `typeinfo` is not available.

Reply:
Hm, never mind I guess; in this branch we don't print any field type info.